File: LeetCode/C#/1588. Sum of All Odd Length Subarrays.cs
public class Solution {
    public int SumOddLengthSubarrays(int[] arr) {
        if(arr.Length == 1)
        {
            return arr[0];
        }
        else if(arr.Length == 2)
        {
            return arr[0] + arr[1];
        }
        else
        {
            var sum = 0;
            
            for (int i = 0; i < arr.Length; i++)
            {
                var currentSum = 0;
                var x = 1;
                
                for (int j = i; j < arr.Length; j++)
                {
                    currentSum += arr[j];
                    
                    if(x % 2 != 0)
                    {
                        sum += currentSum;
                    }
                    
                    x++;
                }
            }
            
            return sum;
        }
    }
}

File: LeetCode/C++/1588. Sum of All Odd Length Subarrays.cpp
class Solution
{
public:
    int sumOddLengthSubarrays(vector<int> &arr)
    {
        int sum = 0;
        int n = arr.size();

        for (int i = 0; i < n; i++)
        {
            int occurrences = ceil((i + 1) * (n - i) / 2.0);
            sum += arr[i] * occurrences;
        }

        return sum;
    }
};